### ISSUE TITLE
InvalidOperationException

### ISSUE THREAD

**Issue by johnnyontheweb** [mkaring/ConfuserEx#122]
Hello all,
I get this error when running 1.3.1 on a NET 4 DLL compiled with /CLR.

> [ERROR] Unknown error occurred.
Exception: System.InvalidOperationException: Operation not valid due to the current status of the object.
   in dnlib.DotNet.Writer.MethodBodyChunks.WriteReusedMethodBodies(DataWriter writer, Int64 destStreamBaseOffset)
   in dnlib.DotNet.Writer.NativeModuleWriter.WriteFile()
   in dnlib.DotNet.Writer.NativeModuleWriter.WriteImpl()
   in dnlib.DotNet.Writer.ModuleWriterBase.Write(Stream dest)
   in Confuser.Core.ConfuserEngine.WriteModule(ConfuserContext context)
   in Confuser.Core.ProtectionPipeline.ExecuteStage(PipelineStage stage, Action`1 func, Func`1 targets, ConfuserContext context)
   in Confuser.Core.ConfuserEngine.RunPipeline(ProtectionPipeline pipeline, ConfuserContext context)
   in Confuser.Core.ConfuserEngine.RunInternal(ConfuserParameters parameters, CancellationToken token)
Failed at 09:54, 0:23 elapsed.

Can you help on this? any hints?

**Comment by mkaring** [comment]
I never tested ConfuserEx with C++/CLR assemblies. It's only meant to handle pure managed assemblies and not mixed mode assenblies. This error looks like a problem when writing the assembly back out so I am guessing the problem is Confuser either corrupting the assembly or dnlib not reading it correctly.

The question is actually why you want to obfuscate the assembly. The optimization of the C++ compiler should make the code sufficiently unreadable.

**Comment by johnnyontheweb** [comment]
Yes, no problem. I can use ConfuserEx 1.0 (from the original project) to get this.
The error sound strange to me too, it seems there's a writing problem but, as I said, previous version works perfectly.

**Comment by mkaring** [comment]
Okay. I'll honestly can't tell you what goes wrong here. I'll need to setup a test project and see what's what. Is there any chance you could provide a minimal example producing the issue?

**Comment by mkaring** [comment]
I looked in some more detail into this issue and found the tamper protection to be the only one leading to this error. Currently that protection does not work with the C++/CLI assembly, all others work fine it seems. I only used a very simple project as reference for the tests, so it may be different for you. It's a thing you can try in order to work around your issue.

**Comment by johnnyontheweb** [comment]
Thank you so much for your help, I'll try to exclude tamper protection

**Comment by johnnyontheweb** [comment]
Excluding tamper protection worked, thanks. I'm wondering why there was no need to exclude it with ConfuserEx 1.0. can you tell something on that?
